modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultIRTreeToASMBytesPhase.java

```diff
@@ -1763,4 +1784,7 @@ public void visitDup(DupNode irDupNode, WriteScope writeScope) {
 
         methodWriter.writeDup(size, depth);
     }
+
+    // placeholder class referring to the script instance
+    private static final class $this {}
```

Comment:
I'd probably name this a standard java class name to not confuse it with the debugger output for the internal variable. It's like naming a class `_` or something. Its probably legal but probably confusing.

Reply:
Done.